sympy/printing/tests/test_numpy.py

```diff
@@ -252,3 +252,21 @@ def test_16857():
 
     printer = NumPyPrinter()
     assert printer.doprint(A) == 'numpy.block([[a_1, a_2], [a_3, a_4]])'
+
+
+def test_issue_17006():
+    if not np:
+        skip("NumPy not installed")
+
+    M = MatrixSymbol("M", 2, 2)
+
+    f = lambdify(M, M + Identity(2))
+    ma = np.array([[1, 2], [3, 4]])
+    mr = np.array([[2, 2], [3, 5]])
+
+    assert (f(ma) == mr).all()
+
+    from sympy import symbols
+    n = symbols('n', integer=True)
+    N = MatrixSymbol("M", n, n)
+    raises(NotImplementedError, lambdify(N, N + Identity(n)))
```

Comment:
You are using raises incorrectly here. You need to give raises a function that it calls otherwise the error comes from the expression for the second argument before the raises function is called. You either use raises like this
```
raises(NotImplementedError, lambda: lambdify(N, N + Identity(n)))
```
or like this
```
with raises(NotImplementedError):
    lambdify(N, N + Identity(n)))
```